core/src/test/scala/io/chrisdavenport/curly/MainSpec.scala

```diff
@@ -7,39 +7,48 @@ class MainSpec extends munit.FunSuite {
   test("leading 'curl' is case insensitive") {
     val cmd = "CuRl 'https://typelevel.org/'"
     val actual = Curly.all.parseAll(cmd)
-    // TODO actually a URI
     val expected = List(Unhandled("https://typelevel.org/"))
     assertEquals(actual, Right(expected))
   }
 
   test("only a URL") {
     val cmd = "curl 'https://typelevel.org/'"
     val actual = Curly.all.parseAll(cmd)
-    // TODO actually a URI
     val expected = List(Unhandled("https://typelevel.org/"))
     assertEquals(actual, Right(expected))
   }
 
-  //
-  // Just asserting lengths for now...
-  //
-
   test("with one header") {
     val cmd = "curl 'https://typelevel.org/' -H 'User-Agent: Mozilla/5.0'"
     val actual = Curly.all.parseAll(cmd)
-    assertEquals(actual.map(_.length), Right(2))
+    val expected = List(
+      Unhandled("https://typelevel.org/"),
+      Opt("header", "User-Agent: Mozilla/5.0"),
+    )
+    assertEquals(actual, Right(expected))
   }
 
   test("with two headers") {
     val cmd = "curl 'https://typelevel.org/' -H 'User-Agent: Mozilla/5.0' -H 'Connection: keep-alive'"
     val actual = Curly.all.parseAll(cmd)
-    assertEquals(actual.map(_.length), Right(3))
+    val expected = List(
+      Unhandled("https://typelevel.org/"),
+      Opt("header", "User-Agent: Mozilla/5.0"),
+      Opt("header", "Connection: keep-alive"),
+    )
+    assertEquals(actual, Right(expected))
   }
 
-  test("with two headers separated by a long arg") {
+  test("with two headers separated by a long arg".fail) {
```

Comment:
@ChristopherDavenport How about this for handling failing tests for now?
This asserts that it *should* fail.
And then if we fix this test case it'll instead start failing (as in failing to fail) and we can drop the `.fail`

Reply:
Why would this fail? Its allowed to do that.

Reply:
Sorry, I wasn't very clear there.
The assertion currently fails because `--compress` actually ends up as `Unhandled("--compress")` instead of `Flag("compress")`.

But I don't want that to fail CI.
So I've encoded this as a failing test. Which means the test passes only if the assertion fails.

And then when we fix the parser to return `Flag("compress")` this whole test will start failing and we'll remove the `.fail` and have nice passing test again.

Does that make sense?

Reply:
Ah, got it.